### DISCUSSION TITLE
How to get nvim-tree to play nice when deleting the current buffer?

### DISCUSSION THREAD

**Discussion by unphased** [nvim-tree/nvim-tree.lua#2667] [Q&A] [answered]
I like to use the bufline in my statusline plugin to navigate through buffers in nvim. This mode of working works really well with nvim-tree in a way that doesn't with many other file tree plugins, so props for that. 

However when I want to delete the current buffer in order to eliminate it from cluttering up the vim buflist, this causes nvimtree to remain and fill up the window, when what I would want is for the next buffer in the list to become active. 

Probably I can implement some janky workaround, but given that this works perfectly whenever nvimtree is not active, I wonder if there is a slight tweak to functionality we could do inside nvimtree?

This is very related to the issue of trying to close nvim (`:q`) when nvimtree is open. it leaves nvimtree fullscreen instead of closing nvim.

**Comment by gegoune** [comment]
It is indeed very related, almost the same. You can check out [this solution](https://github.com/nvim-tree/nvim-tree.lua/wiki/Recipes#go-to-last-used-hidden-buffer-when-deleting-a-buffer).

**Reply by unphased** [reply]
Hardly a more appropriate circumstance for the heart emoji. Much thanks.

**Reply by unphased** [reply]
Alright, interesting. So with this installed: 

- `:q` won't quit vim anymore, it just ... does nothing as the autocmd brings the last location back so it's in the same state. This might be ok, as I'd be happy to just get used to `:qa` to quit. The only reason this isn't entirely satisfactory is the new inconsistency of `:q` still quitting vim, if I *don't* have nvimtree open... That said, if I do go toward `:qa` to quit with intention, it also would paper over this inconsistency... I might be able to convince myself it's not inconsistent and having nvimtree open merely makes vim stickier to another degree.
- buf delete with nvimtree open opens an empty buffer. This is way better than leaving nvimtree as full size, because this preserves nvimtree's positioning and leaves a clean slate to reopen more buffers with, so this is not a negative. That said though. in the more likely scenario where i would want to open a file at this point instead of edit an empty blank non-file buffer, the original behavior has one aspect that's better which is that the cursor is in nvimtree when that happens...

Overall it's an improvement. It's just not completely perfect. I have no doubt once I see a bit better how I'd like it to behave, I could tweak the code to do something I'll be more happy with.

**Reply by alex-courtis** [reply, marked as answer]
I'm still using the old but good plugin [vim-bufkill](https://github.com/qpkorr/vim-bufkill)

See these mapped functions to wipe current or all buffers without closing the window: https://github.com/alex-courtis/arch/blob/d7f5218bc60ced24a8b4dc22d9c11c374d6b8f18/config/nvim/lua/amc/buffers.lua#L117

**Reply by unphased** [reply]
@alex-courtis I finally got to trying that out and it works great in this situation. Much better experience than that autocommand.
